rocks/RocksProperties.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.core.rocks;
+
+import org.rocksdb.OptimisticTransactionDB;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.TableProperties;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.function.Function;
+
+import static com.vaticle.typedb.core.common.collection.Bytes.MB;
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Internal.STORAGE_PROPERTY_EXCEPTION;
+
+/**
+ * A list of interesting properties to retrieve from RocksDB.
+ * All properties are visible here:
+ * https://github.com/facebook/rocksdb/blob/20357988345b02efcef303bc274089111507e160/include/rocksdb/db.h#L750
+ */
+public enum RocksProperties {
```

Comment:
this class defines interesting properties we may want to Log periodically from RocksDB -
there are two types:
- runtime properties (rocksDB terminology), which are runtime properties like cache usage and compaction bytes to process
- data properties, which I define as an aggregated summary of the data that is persisted in SST on disk, such as total size of bloom filters, indexes, data etc.